File: PropertyGrid/VB/FormPropertyValuesTranslationMethod1.vb
Public Class FormPropertyValuesTranslationMethod1

    ' By handling AdvPropertyGrid events you can perform "bulk" property value conversions
    ' For example apply it to all boolean properties.
    Private Const Text_Yes As String = "Yes"
    Private Const Text_No As String = "No"

    Private Sub FormPropertyValuesTranslationMethod1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        advPropertyGrid1.SelectedObject = buttonX1
    End Sub

    Private Sub advPropertyGrid1_ConvertFromStringToPropertyValue(ByVal sender As Object, ByVal e As DevComponents.DotNetBar.ConvertValueEventArgs) Handles advPropertyGrid1.ConvertFromStringToPropertyValue
        If e.PropertyDescriptor.PropertyType Is GetType(Boolean) Then
            If e.StringValue = Text_Yes Then
                e.TypedValue = True
            Else
                e.TypedValue = False
            End If
            ' Note that you must indicate that you have converted the value
            e.IsConverted = True
        End If
    End Sub

    Private Sub advPropertyGrid1_ConvertPropertyValueToString(ByVal sender As Object, ByVal e As DevComponents.DotNetBar.ConvertValueEventArgs) Handles advPropertyGrid1.ConvertPropertyValueToString
        If e.PropertyDescriptor.PropertyType Is GetType(Boolean) Then
            If e.TypedValue = True Then
                e.StringValue = Text_Yes
            Else
                e.StringValue = Text_No
            End If
            ' Note that you must indicate that you have converted the value
            e.IsConverted = True
        End If
    End Sub

    Private Sub advPropertyGrid1_ProvidePropertyValueList(ByVal sender As Object, ByVal e As DevComponents.DotNetBar.PropertyValueListEventArgs) Handles advPropertyGrid1.ProvidePropertyValueList
        If e.PropertyDescriptor.PropertyType Is GetType(Boolean) Then
            ' Text representation of all available property values
            e.ValueList = New List(Of String)(New String() {Text_Yes, Text_No})
            ' Indicate that a list was provided
            e.IsListValid = True
        End If
    End Sub
End Class

File: PropertyGrid/CS/FormPropertyValuesTranslationMethod1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevComponents.DotNetBar;

namespace PropertyGrid
{
    public partial class FormPropertyValuesTranslationMethod1 : DevComponents.DotNetBar.Office2007Form
    {
        public FormPropertyValuesTranslationMethod1()
        {
            InitializeComponent();
        }

        // By handling AdvPropertyGrid events you can perform "bulk" property value conversions
        // For example apply it to all boolean properties.
        private const string Text_Yes = "Yes";
        private const string Text_No = "No";
        
        // Convert typed property value to its text representation
        private void advPropertyGrid1_ConvertPropertyValueToString(object sender, ConvertValueEventArgs e)
        {
            if (e.PropertyDescriptor.PropertyType == typeof(bool))
            {
                if ((bool)e.TypedValue == true)
                    e.StringValue = Text_Yes;
                else
                    e.StringValue = Text_No;
                // Note that you must indicate that you have converted the value
                e.IsConverted = true;
            }
        }

        // Convert the property value text representation to typed value
        private void advPropertyGrid1_ConvertFromStringToPropertyValue(object sender, ConvertValueEventArgs e)
        {
            if (e.PropertyDescriptor.PropertyType == typeof(bool))
            {
                if (e.StringValue == Text_Yes)
                    e.TypedValue = true;
                else
                    e.TypedValue = false;
                // Note that you must indicate that you have converted the value
                e.IsConverted = true;
            }
        }

        // Provide list of all available property values in text representation, if applicable
        private void advPropertyGrid1_ProvidePropertyValueList(object sender, PropertyValueListEventArgs e)
        {
            if (e.PropertyDescriptor.PropertyType == typeof(bool))
            {
                // Text representation of all available property values
                e.ValueList = new List<string>(new string[] { Text_Yes, Text_No });
                // Indicate that a list was provided
                e.IsListValid = true;
            }
        }

        private void FormPropertyValuesTranslationMethod1_Load(object sender, EventArgs e)
        {
            advPropertyGrid1.SelectedObject = buttonX1;
        }
    }
}